ray_on_golem/ctl/ctl.py

```diff
@@ -130,29 +147,92 @@ def stop_webserver(
 
         return shutdown_state
 
-    def wait_for_shutdown(self) -> None:
+    def wait_for_stop(
+        self,
+        starting_up=True,
+        shutdown_timeout=RAY_ON_GOLEM_SHUTDOWN_TIMEOUT,
+        stop_timeout=RAY_ON_GOLEM_STOP_TIMEOUT,
+    ):
+        self._wait_for_shutdown(starting_up=starting_up, timeout=shutdown_timeout)
+        self._wait_for_process_stop(starting_up=starting_up, timeout=stop_timeout)
+
+    def _wait_for_shutdown(
+        self,
+        starting_up: bool,
+        timeout=RAY_ON_GOLEM_SHUTDOWN_TIMEOUT,
+    ) -> None:
+        webserver = "Previous webserver instance" if starting_up else "Webserver"
+        if self._client.is_webserver_serviceable() is None:
+            return
+
         self._output_logger.info(
-            "Previous webserver instance is still shutting down, "
-            f"waiting with deadline up to `{RAY_ON_GOLEM_SHUTDOWN_DEADLINE}`...",
+            f"{webserver} is still shutting down, "
+            f"waiting up to {int(timeout.total_seconds())} seconds ...",
         )
 
-        wait_deadline = datetime.now() + RAY_ON_GOLEM_SHUTDOWN_DEADLINE
-        check_seconds = int(RAY_ON_GOLEM_CHECK_DEADLINE.total_seconds())
+        wait_deadline = datetime.now() + timeout
+        check_seconds = int(RAY_ON_GOLEM_CHECK_INTERVAL.total_seconds())
 
-        time.sleep(check_seconds)
         while datetime.now() < wait_deadline:
+            time.sleep(check_seconds)
             webserver_serviceable = self._client.is_webserver_serviceable()
             if webserver_serviceable is None:
-                self._output_logger.info("Previous webserver instance shutdown done")
+                self._output_logger.info(f"{webserver} shutdown completed.")
                 return
 
-            self._output_logger.info(
-                "Previous webserver instance is not yet shutdown, "
-                f"waiting additional `{check_seconds}` seconds..."
+            self._output_logger.verbose(
+                f"{webserver} is still running, " f"waiting additional `{check_seconds}` seconds..."
             )
+
+        self._output_logger.error(f"Shutdown timeout of {timeout} reached.")
+
+    def _wait_for_process_stop(self, starting_up=True, timeout=RAY_ON_GOLEM_STOP_TIMEOUT):
```

Comment:
As this method is used only in one place, I think that the default value is no longer needed, as it will be provided by the only callee wait_for_stop().

Reply:
:+1: